src/airflow_tools/providers/filesystem/operators/filesystem.py

```diff
@@ -53,3 +54,58 @@ def execute(self, context):
             else self.destination_path
         )
         destination_fs_hook.write(data, full_destination_path)
+
+
+class SQLToFilesystem(BaseOperator):
+    """
+    Copies data from a SQL query to a filesystem.
+    """
+
+    template_fields = (
+        'source_name',
+        'source_schema_name',
+        'source_table_name',
+        'destination_path',
+    )
+
+    def __init__(
+        self,
+        source_sql_conn_id: str,
+        destination_fs_conn_id: str,
+        source_name: str,
+        source_schema_name: str,
+        source_table_name: str,
+        destination_path: str,
+        batch_size: Optional[int] = 100000,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.source_sql_conn_id = source_sql_conn_id
+        self.destination_fs_conn_id = destination_fs_conn_id
+        self.source_name = source_name
+        self.source_schema_name = source_schema_name
+        self.source_table_name = source_table_name
+        self.destination_path = destination_path
+        self.batch_size = batch_size
+
+    def execute(self, context):
+        source_sql_hook: DbApiHook = BaseHook.get_connection(
+            self.source_sql_conn_id
+        ).get_hook()
+        source_sql_hook.__schema__ = self.source_schema_name
```

Comment:
I would probably prefer to let the hook's connection params decide the schema and not take it as a parameter. This will make it easier to test (we can point it to a test schema without changing the DAG parameters).